Services/Worker/OJS.Services.Worker.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -148,7 +148,9 @@ private static void PreprocessSubmission(SubmissionServiceModel submission)
         var ojsWorkerExecutionResult = this.submissionExecutor.Execute<TInput, TResult>(ojsSubmission);
 
         var executionResult = this.mapper.Map<ExecutionResultServiceModel>(ojsWorkerExecutionResult);
+
         executionResult.StartedExecutionOn = ojsSubmission.StartedExecutionOn;
+        executionResult.CompletedExecutionOn = DateTime.UtcNow;
```

Comment:
Lets move CompletedExecutionOn in the end of the execution, just before returning the result and make sure StartedExecutionOn is at the beggining.

Reply:
Done.